### DISCUSSION TITLE
Do some Taxes Freezes and doesn't finish calculating

### DISCUSSION THREAD

**Discussion by zigza123** [davetapley/helium-tax#81] [General]
I am assuming this is because #79 but unsure if anyone else is having the same problem.

**Comment by ArizonaDaddio** [comment]
Same problem here.

**Reply by kwatkins99** [reply]
Same problem here.

**Reply by Micheal-McLaughlin** [reply]
Same problem here - get's to about 197 transactions and then stops - Have you rate limited it?

**Reply by davetapley** [reply]
@Micheal-McLaughlin I've tried slowing it down very aggressively, but the Helium API still starts return HTTP 429.
I am working on an different approach right now (see #83).

**Comment by schnub69** [comment]
+1

**Comment by davetapley** [comment]
Yep I think it's safe to assume this is all #79.
I have some ideas for further workarounds which I'll work on later today.

Please subscribe to #79 for further updates, but you're welcome to keep trying in the mean time.

**Comment by davetapley** [comment]
Okay I've got a fix in via https://github.com/davetapley/helium-tax/issues/83.
It limits us to just 2020 and 2021, but should at least let everyone do their 2021 taxes.

Please spot check a few rows and **let me know ASAP if you see anything wrong**
(preferably with hotspot/account info and row number etc.) 🙏🏻
